lib/jekyll/drops/drop.rb

```diff
@@ -209,6 +209,18 @@ def fetch(key, default = nil, &block)
         return yield(key) unless block.nil?
         return default unless default.nil?
       end
+
+      private
+      def marshal_dump
+        (instance_variables - [:@context]).map do |attr|
```

Comment:
How may times will this method get called in an average-to-large site?
We can reduce Array allocations by using `map!` here, since `Array#-` already creates a copy of given array, `instance_variables`)

Reply:
It's going to be called at most once for every drop. I just checked it with the jekyll docs site:
```ruby
ObjectSpace.each_object(Jekyll::Drops::Drop).count # => 257
page = ObjectSpace.each_object(Jekyll::Page).first
Marshal.dump(page) # `marshal_dump` in `Jekyll::Drops::Drop` is called 252 times
```

Thanks for the suggestion with `#map!` 👍  i updated the code